opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -35,14 +35,33 @@ class ControllerModuleSmailyForOpencart extends Controller {
     public function index() {
         // Add language file.
         $this->load->language('module/smaily_for_opencart');
-        // Smaily admin page model.
+        // Settings model
+        $this->load->model('setting/setting');
+        // Smaily admin model
         $this->load->model('smailyforopencart/admin');
-        $settings = $this->load->model_smailyforopencart_admin->getSettingValue('smaily', 'smaily_api_credentials'); 
+        // Load smaily settings
+        $settings = $this->load->model_setting_setting->getSetting('smaily'); 
         // Add heading title.
         $this->document->setTitle($this->language->get('heading_title'));
+        // When save is pressed.
+        if (($this->request->server['REQUEST_METHOD'] == 'POST')) {
+            // Save Customer Sync settings.
+            $this->handleCustomerSync();
+            // Display Success after pressing save.
+            $this->session->data['success'] = $this->language->get('text_success');
+            // Redirect to module settings page.
+            $this->redirect($this->url->link('extension/module', 'token=' . $this->session->data['token'], 'SSL'));
```

Comment:
Add `return` before `$this->redirect`. It does not have any effect, but will signal person, without knowledge to how `redirect` method works, to know that code execution stops here.